### PR TITLE
Refactored code to make Sherlock a proper CLI tool and python package

### PR DESCRIPTION
**Pull request by avinashshenoy97**
1. Changed code structure for to be a python package - can create setup.py and register on PyPI to make sherlock pip installable

2. Moved Sherlock to a class that needs to be instantiated with a username. This sherlock object can be used to check that username on some/all supported sites

3. Prints and CLI features moved to __main__.py so that all the findings are printed only when "python sherlock --site <site> <username>" is run.

4. The site arg can now be "facebook" or even just "face" and it will be matched to facebook because of how the argument is compared to the name of the websites on the supported sites list.

5. Moved helper functions to watson.py (hehe)

6. Adding a "loading /" line that shows up while Sherlock analyzes the given username's presence across all the sites.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,32 +43,38 @@ optional arguments:
...
                         Display extra debugging information and metrics.
-  --rank, -r            Present websites ordered by their Alexa.com global
-                        rank in popularity.
-  --folderoutput FOLDEROUTPUT, -fo FOLDEROUTPUT
-                        If using multiple usernames, the output of the results
-                        will be saved at this folder.
-  --output OUTPUT, -o OUTPUT
-                        If using single username, the output of the result
-                        will be saved at this file.
   --tor, -t             Make requests over TOR; increases runtime; requires
```
Comment (ptalmeida) [inline README.md]: I don't get why --rank, --folderoutput  and --json were removed

**Comment by hoadlck** [comment]
This is the 3rd pull request for doing the restructuring.  I would like to get from @TheYahya on what is the way forward on these? It does not seem sensible to have multiple people doing the same thing.

File: README.md (hunk 1 above)
Comment (avinashshenoy97) [reply README.md]: Slight oversight on my part... added them back :)

**Comment by avinashshenoy97** [comment]
The checks are failing because of a connection error... it's a transient error. If the build is triggered again, it will probably go through.

**Comment by TheYahya** [comment]
Thank you @avinashshenoy97, 
I prefer the way @knijam did at #112 @hoadlck, but it has so many changes and there's so conflicts too. 
And I check all of them and compare them.

**Comment by hoadlck** [comment]
Much of the goals in this has been incorporated into #590.

**Pull request closed without merging**